cli/azd/resources/scaffold/base/shared/apps-env.bicep

```diff
@@ -0,0 +1,33 @@
+param name string
```

Comment:
I looked at the generated code.  Personally, I feel that the second approach (without core) is much nicer.  There are far fewer files and each individual file feels much more approachable.  For someone who is new to IaC and 
Azure, it feels a lot less scary when I look at the second one and the platform as a whole feels way less complex. This simplification bleeds across the platform - deployments will now be less deep, making exploring what's going on in the Portal much nicer.  You won't be clicking down a complex tree view to see the actual CRUD operations.

While we just landed changes to include the source for the core modules in the starter templates, I feel like not using them here still makes sense.  We have a lot more context and can generate bespoke bicep that the user can maintain.  We can have much smarter defaults because they can be informed by app context and powered by a rich programming language.  For folks that need to go further, they have the full power of the platform instead of needing our modules to expose every nob.  In a world where Copilot is becoming more and more popular, I also think that just using raw bicep instead of our modules may lead to improved AI experiences because the model has seen so much more raw bicep vs "azd flavored module bicep"

Reply:
Considering both options here, I agree in spirit with @ellismg and @weikanglim. I think that for our target audience for the feature going with the smaller footprint, generated code here is the best UX. I do think that core is hugely valuable and as we increase the fidelity of this offer beyond alpha, we should continue figure out how we can enable developers to grow up their applications. I think that for v1 and for the zero-state scenario this ends up probably being a bit heavier weight than we'd like and would likely be overwhelming and feel potentially unapproachable to new users.

Remember, this is an experimental/alpha stage feature and we **will** end up making changes here.

That said, my recommendation is:
- Generate the bespoke assets that are as simple as possible
- Do not include core in infra
- Add a note to the generated next-steps.md to point developers at the starter Bicep template (which will include the core modules in v1.3.0) and explain how they can leverage to to make changes/grow up their app.

Reply:
Thanks for all the feedback. I'll continue with the current implementation for now. We'll reevaluate and readjust as we learn more from our users in experimental/alpha stage. 

I've kept a stashed copy of the `core` implementation in https://github.com/weikanglim/azure-dev/tree/weilim/scaffold-infra-core for future reference.